File: backend/app/models/xhs.py
"""
小红书批量图文笔记相关模型。
"""
from datetime import datetime
from typing import TYPE_CHECKING, Optional

from sqlalchemy import (
    Boolean,
    DateTime,
    ForeignKey,
    Integer,
    Numeric,
    String,
    Text,
    UniqueConstraint,
)
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base, TimestampMixin
from app.models.types import JSONType

if TYPE_CHECKING:
    from app.models.tenant import Tenant


class XHSRulePack(Base, TimestampMixin):
    __tablename__ = "xhs_rule_packs"
    __table_args__ = (
        UniqueConstraint("tenant_id", "category_id", "version", name="uq_xhs_rule_pack_version"),
    )

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    tenant_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("tenants.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    category_id: Mapped[str] = mapped_column(String(64), nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    version: Mapped[str] = mapped_column(String(64), nullable=False)
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="draft", index=True)
    pack_json: Mapped[dict] = mapped_column(JSONType, nullable=False)
    created_by: Mapped[str] = mapped_column(String(64), nullable=False)

    tenant: Mapped["Tenant"] = relationship("Tenant", back_populates="xhs_rule_packs")


class XHSProject(Base, TimestampMixin):
    __tablename__ = "xhs_projects"

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    tenant_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("tenants.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    category_id: Mapped[str] = mapped_column(String(64), nullable=False, index=True)
    client_name: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
    product_name: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
    brief_file_ref: Mapped[Optional[str]] = mapped_column(String(2048), nullable=True)
    brief_file_name: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
    brief_parse_result_json: Mapped[Optional[dict]] = mapped_column(JSONType, nullable=True)
    project_brief: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    shared_requirements: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    remark: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="active", index=True)
    created_by: Mapped[str] = mapped_column(String(64), nullable=False)

    tenant: Mapped["Tenant"] = relationship("Tenant", back_populates="xhs_projects")
    variants: Mapped[list["XHSProjectVariant"]] = relationship(
        "XHSProjectVariant",
        back_populates="project",
        cascade="all, delete-orphan",
        lazy="selectin",
    )
    directions: Mapped[list["XHSDirectionItem"]] = relationship(
        "XHSDirectionItem",
        back_populates="project",
        cascade="all, delete-orphan",
        lazy="selectin",
    )


class XHSProjectVariant(Base, TimestampMixin):
    __tablename__ = "xhs_project_variants"
    __table_args__ = (
        UniqueConstraint("project_id", "name", name="uq_xhs_project_variant_name"),
    )

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    tenant_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("tenants.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    project_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("xhs_projects.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    selling_points: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    appearance_notes: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    notes: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    is_primary: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    sort_order: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    created_by: Mapped[str] = mapped_column(String(64), nullable=False)

    tenant: Mapped["Tenant"] = relationship("Tenant", back_populates="xhs_project_variants")
    project: Mapped["XHSProject"] = relationship("XHSProject", back_populates="variants")
    main_directions: Mapped[list["XHSDirectionItem"]] = relationship(
        "XHSDirectionItem",
        back_populates="main_variant",
        foreign_keys="XHSDirectionItem.main_variant_id",
        lazy="selectin",
    )


class XHSDirectionItem(Base, TimestampMixin):
    __tablename__ = "xhs_direction_items"
    __table_args__ = (
        UniqueConstraint("project_id", "name", name="uq_xhs_direction_item_name"),
    )

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    tenant_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("tenants.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    project_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("xhs_projects.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="draft", index=True)
    main_variant_id: Mapped[Optional[str]] = mapped_column(
        String(64),
        ForeignKey("xhs_project_variants.id", ondelete="SET NULL"),
        nullable=True,
        index=True,
    )
    secondary_variant_ids_json: Mapped[Optional[list]] = mapped_column(JSONType, nullable=True)
    content_style: Mapped[Optional[str]] = mapped_column(String(64), nullable=True)
    direction_brief: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    extra_requirements: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    notes: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    sort_order: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    created_by: Mapped[str] = mapped_column(String(64), nullable=False)

    tenant: Mapped["Tenant"] = relationship("Tenant", back_populates="xhs_direction_items")
    project: Mapped["XHSProject"] = relationship("XHSProject", back_populates="directions")
    main_variant: Mapped[Optional["XHSProjectVariant"]] = relationship(
        "XHSProjectVariant",
        back_populates="main_directions",
        foreign_keys=[main_variant_id],
    )
    batches: Mapped[list["XHSBatchJob"]] = relationship(
        "XHSBatchJob",
        back_populates="direction",
        lazy="selectin",
    )


class XHSBrandPack(Base, TimestampMixin):
    __tablename__ = "xhs_brand_packs"
    __table_args__ = (
        UniqueConstraint("tenant_id", "category_id", "version", name="uq_xhs_brand_pack_version"),
    )

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    tenant_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("tenants.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    brand_name: Mapped[str] = mapped_column(String(255), nullable=False)
    category_id: Mapped[str] = mapped_column(String(64), nullable=False, index=True)
    version: Mapped[str] = mapped_column(String(64), nullable=False)
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="draft", index=True)
    is_default: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    pack_json: Mapped[dict] = mapped_column(JSONType, nullable=False)
    created_by: Mapped[str] = mapped_column(String(64), nullable=False)

    tenant: Mapped["Tenant"] = relationship("Tenant", back_populates="xhs_brand_packs")


class XHSBriefPack(Base, TimestampMixin):
    __tablename__ = "xhs_brief_packs"
    __table_args__ = (
        UniqueConstraint("tenant_id", "category_id", "version", name="uq_xhs_brief_pack_version"),
    )

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    tenant_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("tenants.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    brand_name: Mapped[str] = mapped_column(String(255), nullable=False)
    category_id: Mapped[str] = mapped_column(String(64), nullable=False, index=True)
    version: Mapped[str] = mapped_column(String(64), nullable=False)
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="draft", index=True)
    pack_json: Mapped[dict] = mapped_column(JSONType, nullable=False)
    source_type: Mapped[str] = mapped_column(String(32), nullable=False)
    source_ref: Mapped[Optional[str]] = mapped_column(String(2048), nullable=True)
    created_by: Mapped[str] = mapped_column(String(64), nullable=False)

    tenant: Mapped["Tenant"] = relationship("Tenant", back_populates="xhs_brief_packs")


class XHSRiskPack(Base, TimestampMixin):
    __tablename__ = "xhs_risk_packs"
    __table_args__ = (
        UniqueConstraint("tenant_id", "category_id", "version", name="uq_xhs_risk_pack_version"),
    )

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    tenant_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("tenants.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    category_id: Mapped[str] = mapped_column(String(64), nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    version: Mapped[str] = mapped_column(String(64), nullable=False)
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="draft", index=True)
    pack_json: Mapped[dict] = mapped_column(JSONType, nullable=False)
    created_by: Mapped[str] = mapped_column(String(64), nullable=False)

    tenant: Mapped["Tenant"] = relationship("Tenant", back_populates="xhs_risk_packs")


class XHSBatchJob(Base, TimestampMixin):
    __tablename__ = "xhs_batch_jobs"

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    tenant_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("tenants.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    created_by: Mapped[str] = mapped_column(String(64), nullable=False, index=True)
    status: Mapped[str] = mapped_column(String(32), nullable=False, default="pending", index=True)
    category_id: Mapped[str] = mapped_column(String(64), nullable=False, index=True)
    direction_id: Mapped[Optional[str]] = mapped_column(
        String(64),
        ForeignKey("xhs_direction_items.id", ondelete="SET NULL"),
        nullable=True,
        index=True,
    )
    rule_pack_version: Mapped[Optional[str]] = mapped_column(String(64), nullable=True)
    risk_pack_version: Mapped[Optional[str]] = mapped_column(String(64), nullable=True)
    brand_pack_version: Mapped[Optional[str]] = mapped_column(String(64), nullable=True)
    brief_pack_id: Mapped[Optional[str]] = mapped_column(String(64), nullable=True)
    style_template_id: Mapped[Optional[str]] = mapped_column(String(64), nullable=True)
    run_mode: Mapped[str] = mapped_column(String(20), nullable=False, default="trial")
    trial_sample_count: Mapped[Optional[int]] = mapped_column(Integer, nullable=True)
    input_type: Mapped[str] = mapped_column(String(32), nullable=False)
    input_stats_json: Mapped[Optional[dict]] = mapped_column(JSONType, nullable=True)
    tag_policy_json: Mapped[Optional[dict]] = mapped_column(JSONType, nullable=True)
    export_options_json: Mapped[Optional[dict]] = mapped_column(JSONType, nullable=True)
    estimated_tokens: Mapped[Optional[int]] = mapped_column(Integer, nullable=True)
    estimated_cost: Mapped[Optional[float]] = mapped_column(Numeric(12, 4), nullable=True)
    actual_tokens: Mapped[Optional[int]] = mapped_column(Integer, nullable=True)
    actual_cost: Mapped[Optional[float]] = mapped_column(Numeric(12, 4), nullable=True)
    system_blocked: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    system_block_reason: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    total_items: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    done_items: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    running_items: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    export_all_md_status: Mapped[Optional[str]] = mapped_column(String(32), nullable=True)
    export_all_md_url: Mapped[Optional[str]] = mapped_column(String(2048), nullable=True)
    export_feishu_status: Mapped[Optional[str]] = mapped_column(String(32), nullable=True)
    export_feishu_doc_title: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
    export_feishu_error: Mapped[Optional[str]] = mapped_column(Text, nullable=True)

    tenant: Mapped["Tenant"] = relationship("Tenant", back_populates="xhs_batch_jobs")
    direction: Mapped[Optional["XHSDirectionItem"]] = relationship("XHSDirectionItem", back_populates="batches")
    items: Mapped[list["XHSBatchItem"]] = relationship(
        "XHSBatchItem",
        back_populates="batch",
        cascade="all, delete-orphan",
        lazy="selectin",
    )
    export_logs: Mapped[list["XHSExportLog"]] = relationship(
        "XHSExportLog",
        back_populates="batch",
        cascade="all, delete-orphan",
        lazy="selectin",
    )


class XHSBatchItem(Base):
    __tablename__ = "xhs_batch_items"
    __table_args__ = (
        UniqueConstraint("batch_id", "item_id", name="uq_xhs_batch_item_item_id"),
    )

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    batch_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("xhs_batch_jobs.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    item_id: Mapped[str] = mapped_column(String(64), nullable=False)
    source_text: Mapped[str] = mapped_column(Text, nullable=False)
    source_title_guess: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
    split_by: Mapped[Optional[str]] = mapped_column(String(32), nullable=True)
    status: Mapped[str] = mapped_column(String(32), nullable=False, default="pending", index=True)
    round: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    editor_output_json: Mapped[Optional[dict]] = mapped_column(JSONType, nullable=True)
    verifier_json: Mapped[Optional[dict]] = mapped_column(JSONType, nullable=True)
    verifier_pass: Mapped[Optional[bool]] = mapped_column(Boolean, nullable=True)
    verifier_confidence: Mapped[Optional[float]] = mapped_column(Numeric(5, 4), nullable=True)
    rewrite_fail_reasons_json: Mapped[Optional[list]] = mapped_column(JSONType, nullable=True)
    safe_rewrite_used: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    safe_rewrite_reason: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    final_title: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
    final_body: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    final_hashtags_json: Mapped[Optional[list]] = mapped_column(JSONType, nullable=True)
    copy_ready_text: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    quality_score: Mapped[Optional[float]] = mapped_column(Numeric(5, 2), nullable=True)
    duration_ms: Mapped[Optional[int]] = mapped_column(Integer, nullable=True)
    model_meta_json: Mapped[Optional[dict]] = mapped_column(JSONType, nullable=True)
    started_at: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True)
    finished_at: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True)

    batch: Mapped["XHSBatchJob"] = relationship("XHSBatchJob", back_populates="items")


class XHSExportLog(Base, TimestampMixin):
    __tablename__ = "xhs_export_logs"

    id: Mapped[str] = mapped_column(String(64), primary_key=True)
    batch_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("xhs_batch_jobs.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    type: Mapped[str] = mapped_column(String(32), nullable=False)
    status: Mapped[str] = mapped_column(String(32), nullable=False, default="pending")
    request_json: Mapped[Optional[dict]] = mapped_column(JSONType, nullable=True)
    response_json: Mapped[Optional[dict]] = mapped_column(JSONType, nullable=True)
    error: Mapped[Optional[str]] = mapped_column(Text, nullable=True)

    batch: Mapped["XHSBatchJob"] = relationship("XHSBatchJob", back_populates="export_logs")

File: frontend/types/xhs.ts
export type XHSPackStatus = 'draft' | 'active' | 'archived'
export type XHSProjectStatus = 'active' | 'archived'
export type XHSDirectionStatus = 'draft' | 'active' | 'archived'
export type XHSBatchRunMode = 'trial' | 'full'
export type XHSBatchStatus =
  | 'pending'
  | 'splitting'
  | 'queued'
  | 'running'
  | 'awaiting_decision'
  | 'needs_decision'
  | 'partially_done'
  | 'completed'
  | 'done'
  | 'failed'
  | 'blocked'
  | 'cancelled'
  | 'exporting'
  | 'exported'

export type XHSInputType = 'text' | 'file' | 'feishu_link'
export type XHSExportType = 'all_md' | 'feishu'
export type XHSExportStatus = 'pending' | 'running' | 'completed' | 'failed'
export type XHSSourceType = 'upload' | 'feishu_link'

export interface XHSConfigConflict {
  field: string
  message: string
  severity: string
}

export interface XHSConfigValidationResponse {
  valid: boolean
  conflicts: XHSConfigConflict[]
}

export interface XHSBrandPackPayload {
  brand_facts: Array<Record<string, unknown>>
  products: Array<Record<string, unknown>>
  fact_graph: {
    nodes: Array<Record<string, unknown>>
    relations: Array<Record<string, unknown>>
  }
  optional_blocks: Array<Record<string, unknown>>
}

export interface XHSRulePackPayload {
  banned_terms: string[]
  risk_patterns: Array<Record<string, unknown>>
  replace_map: Record<string, string>
  format_rules: Record<string, unknown>
  structure_rules: Record<string, unknown>
}

export interface XHSBriefPackPayload {
  brand_facts: Record<string, unknown>
  sku_facts: Array<Record<string, unknown>>
  selling_point_priority: Array<Record<string, unknown>>
  recommended_phrasings: string[]
  forbidden_phrasings: string[]
  uncertain_fields: Array<Record<string, unknown>>
}

export interface XHSRiskPackPayload {
  risk_clues: Array<Record<string, unknown>>
  replace_hints: Array<Record<string, unknown>>
  confidence_level?: string | null
}

export interface XHSBrandPack {
  id: string
  tenant_id: string
  category_id: string
  version: string
  status: XHSPackStatus
  created_by: string
  created_at: string
  updated_at: string
  brand_name: string
  is_default: boolean
  pack: XHSBrandPackPayload
}

export interface XHSRulePack {
  id: string
  tenant_id: string
  category_id: string
  version: string
  status: XHSPackStatus
  created_by: string
  created_at: string
  updated_at: string
  name: string
  pack: XHSRulePackPayload
}

export interface XHSBriefPack {
  id: string
  tenant_id: string
  category_id: string
  version: string
  status: XHSPackStatus
  created_by: string
  created_at: string
  updated_at: string
  brand_name: string
  source_type: XHSSourceType
  source_ref?: string | null
  pack: XHSBriefPackPayload
}

export interface XHSRiskPack {
  id: string
  tenant_id: string
  category_id: string
  version: string
  status: XHSPackStatus
  created_by: string
  created_at: string
  updated_at: string
  name: string
  pack: XHSRiskPackPayload
}

export interface XHSBriefPackParseResponse {
  source_type: XHSSourceType
  source_ref?: string | null
  extracted_text: string
  pack: XHSBriefPackPayload
  validation: XHSConfigValidationResponse
}

export interface XHSProjectBriefVariantSuggestion {
  name: string
  selling_points: string[]
  appearance_notes?: string | null
  notes?: string | null
}

export interface XHSProjectBriefDirectionSuggestion {
  name: string
  main_variant_name?: string | null
  secondary_variant_names: string[]
  content_style?: string | null
  direction_brief?: string | null
  extra_requirements: string[]
}

export interface XHSProjectBriefParseResult {
  product_name: string
  project_brief: string
  shared_requirements: string
  key_points: string[]
  variant_suggestions: XHSProjectBriefVariantSuggestion[]
  direction_suggestions: XHSProjectBriefDirectionSuggestion[]
}

export interface XHSProjectBriefParseResponse {
  source_ref: string
  file_name: string
  extracted_text: string
  brief_parse_result: XHSProjectBriefParseResult
  raw_result: Record<string, unknown>
}

export interface XHSVariantBriefParseResult {
  name: string
  selling_points: string[]
  appearance_notes?: string | null
  notes?: string | null
}

export interface XHSVariantBriefParseResponse {
  source_ref?: string | null
  file_name?: string | null
  extracted_text: string
  brief_parse_result: XHSVariantBriefParseResult
  raw_result: Record<string, unknown>
}

export interface XHSProject {
  id: string
  tenant_id: string
  name: string
  category_id: string
  client_name?: string | null
  product_name?: string | null
  brief_file_ref?: string | null
  brief_file_name?: string | null
  brief_parse_result?: XHSProjectBriefParseResult | null
  project_brief?: string | null
  shared_requirements?: string | null
  remark?: string | null
  status: XHSProjectStatus
  created_by: string
  variant_count: number
  direction_count: number
  batch_count: number
  created_at: string
  updated_at: string
}

export interface XHSProjectVariant {
  id: string
  tenant_id: string
  project_id: string
  name: string
  selling_points?: string | null
  appearance_notes?: string | null
  notes?: string | null
  is_primary: boolean
  sort_order: number
  created_by: string
  created_at: string
  updated_at: string
}

export interface XHSDirection {
  id: string
  tenant_id: string
  project_id: string
  project_name?: string | null
  name: string
  status: XHSDirectionStatus
  main_variant_id?: string | null
  main_variant_name?: string | null
  secondary_variant_ids: string[]
  content_style?: string | null
  direction_brief?: string | null
  extra_requirements?: string | null
  notes?: string | null
  sort_order: number
  created_by: string
  batch_count: number
  latest_batch_id?: string | null
  latest_batch_status?: XHSBatchStatus | null
  created_at: string
  updated_at: string
}

export interface XHSBatchEstimateResponse {
  estimated_items: number
  total_split_items: number
  estimated_tokens: number
  estimated_cost: string
  split_strategy: string
}

export interface XHSBatchInputStats {
  raw_chars: number
  split_count: number
  planned_items?: number | null
  split_strategy?: string | null
  split_model?: string | null
  split_tokens?: number | null
  rule_split_count?: number | null
  ai_split_count?: number | null
  source_ref?: string | null
  source_file_name?: string | null
  parsed_from_file?: boolean | null
  parsed_from_feishu?: boolean | null
  parse_skipped_reason?: string | null
}

export interface XHSBatchExportSummary {
  all_md_status?: XHSExportStatus | null
  all_md_url?: string | null
  feishu_status?: XHSExportStatus | null
  feishu_doc_title?: string | null
  feishu_error?: string | null
}

export interface XHSBatchJob {
  id: string
  status: XHSBatchStatus
  category_id: string
  direction_id?: string | null
  direction_name?: string | null
  project_id?: string | null
  project_name?: string | null
  rule_pack_version?: string | null
  risk_pack_version?: string | null
  brand_pack_version?: string | null
  brief_pack_id?: string | null
  run_mode: XHSBatchRunMode
  trial_sample_count?: number | null
  input_type: XHSInputType
  estimated_tokens?: number | null
  estimated_cost?: string | null
  actual_tokens?: number | null
  actual_cost?: string | null
  system_blocked: boolean
  system_block_reason?: string | null
  total_items: number
  done_items: number
  running_items: number
  failed_items: number
  decision_items: number
  safe_rewrite_items: number
  input_stats: XHSBatchInputStats
  export: XHSBatchExportSummary
  export_all_md_status?: XHSExportStatus | null
  export_all_md_url?: string | null
  export_feishu_status?: XHSExportStatus | null
  export_feishu_doc_title?: string | null
  export_feishu_error?: string | null
  created_at: string
  updated_at: string
}

export interface XHSBatchDecisionOption {
  id: string
  title: string
  summary: string
  tradeoffs: string[]
  recommended: boolean
}

export interface XHSBatchItem {
  id: string
  batch_id: string
  item_id: string
  index?: number | null
  status: XHSBatchStatus
  round: number
  title?: string | null
  source_text?: string | null
  source_title_guess?: string | null
  final_title?: string | null
  final_body?: string | null
  final_hashtags: string[]
  copy_ready_text?: string | null
  quality_score?: string | number | null
  verifier_pass?: boolean | null
  verifier_confidence?: string | number | null
  verifier?: Record<string, unknown> | null
  rewrite_fail_reasons?: string[]
  decision_required: boolean
  decision_summary?: string | null
  decision_options: XHSBatchDecisionOption[]
  recommended_decision_option_id?: string | null
  selected_decision_option_id?: string | null
  safe_rewrite_used: boolean
  safe_rewrite_reason?: string | null
  duration_ms?: number | null
}

export interface XHSBatchItemListResponse {
  items: XHSBatchItem[]
  page: number
  page_size: number
  total: number
}

export interface XHSExportLog {
  id: string
  batch_id: string
  type: XHSExportType
  status: XHSExportStatus
  error?: string | null
  created_at: string
}

export interface XHSFeishuExportDoc {
  doc_token: string
  doc_title: string
  doc_url: string
  item_range: string
}

export interface XHSFeishuExportResponse {
  status: XHSExportStatus
  message: string
}

export interface XHSFeishuExportStatusResponse {
  status: XHSExportStatus
  docs: XHSFeishuExportDoc[]
  error?: string | null
}

export interface XHSBatchCreateRequest {
  category_id: string
  direction_id?: string
  rule_pack_version?: string
  risk_pack_version?: string
  brand_pack_version?: string
  brief_pack_id?: string
  style_template_id?: string
  run_mode: XHSBatchRunMode
  trial_sample_count?: number
  input_type: XHSInputType
  input_text?: string
  file_id?: string
  feishu_url?: string
  tag_policy?: Record<string, unknown>
  export_options?: Record<string, unknown>
}
